tests/optimizers/test_general_optimizer.py

```diff
@@ -95,3 +96,14 @@ def test_obj_incorrect_kwargs(self, obj_with_args, optimizer):
         with pytest.raises(TypeError):
             # Wrong kwargs
             cost, pos = optimizer.optimize(obj_with_args, 1000, c=1, d=100)
+
+    def general_eg(self, options):
```

Comment:
Instead of adding a function to generate an optimizer you can just add the parameter `optimizer` in the `test_general_correct_pos` method. For a reference see the test methods above. It'll look like this then:
```python
def test_general_correct_pos(self, options, optimizer):
    """ Sanity test to check if the optimizer returns the position which corresponds to the best cost"""
    cost, pos = optimizer.optimize(sphere, iters=5)
    assert sum(pos ** 2) ==  cost
```
You don't need the `_eg` methods then. The same for the other optimizers.
It may be worth to think about these tests a bit more. Maybe there is a more general (non-circular) check we can apply to see if the optimizer really returns the best position found. Something with the optimizer history for example?

Also, please add a docstring under the function header so it has a short description 😄.